scalalib/src/internal/ModuleUtils.scala

```diff
@@ -0,0 +1,24 @@
+package mill.internal
+
+import mill.define.Module
+import mill.scalalib.JavaModule
+
+@mill.api.internal
+private[mill] object ModuleUtils {
+  // Compute all transitive modules from build children and via moduleDeps
+  def transitiveModules(module: Module, toSkip: Module => Boolean = _ => false): Seq[Module] = {
```

Comment:
I'd prefer a `accept` over a `toSkip`, which default to `true` to include a module. This will also better match semantics of `filter`.